src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -552,12 +555,33 @@ def update_iot_hub_custom(instance,
         instance.properties.cloud_to_device.feedback.max_delivery_count = feedback_max_delivery_count
     if enable_fileupload_notifications is not None:
         instance.properties.enable_file_upload_notifications = enable_fileupload_notifications
+    if fileupload_notification_lock_duration is not None:
+        lock_duration = timedelta(seconds=fileupload_notification_lock_duration)
+        instance.properties.messaging_endpoints['fileNotifications'].lock_duration_as_iso8601 = lock_duration
     if fileupload_notification_max_delivery_count is not None:
         count = fileupload_notification_max_delivery_count
         instance.properties.messaging_endpoints['fileNotifications'].max_delivery_count = count
     if fileupload_notification_ttl is not None:
         ttl = timedelta(hours=fileupload_notification_ttl)
         instance.properties.messaging_endpoints['fileNotifications'].ttl_as_iso8601 = ttl
+    # if setting a fileupload storage identity or changing fileupload to identity-based
+    if fileupload_storage_identity or (fileupload_storage_authentication_type and fileupload_storage_authentication_type.lower() == AuthenticationType.IdentityBased.value):
```

Comment:
Could avoid the `lower()` call in `fileupload_storage_authentication_type.lower()` if `iot/shared.py` attributes had consistent casing with API values.

Reply:
You're right - I think this was done to allow an easier comparison for input casing, can't remember if `get_enum_type` automatically handles input casing or not. Will investigate

Reply:
Ok cool, switching the input type to `get_enum_type(AuthenticationType)` will handle that much better - no more `lower()` checks and it should handle casing / param help as well.